server/src/Metrics.hs

```diff
@@ -1,58 +1,68 @@
+{-# LANGUAGE OverloadedStrings #-}
 module Metrics where
 
-import           Control.Monad         (void)
-import qualified Data.ByteString.Char8 as BS8
-import qualified Network.HTTP.Types    as Http
-import           Prometheus
+import Data.Text (Text, pack)
+import Data.Text.Encoding (decodeUtf8With)
+import Data.Text.Encoding.Error (lenientDecode)
 
-type HttpMethodLabel = String
-type HttpStatusLabel = String
+import qualified Network.HTTP.Types as Http
 
-type HttpRequestCounter = Vector (HttpMethodLabel, HttpStatusLabel) Counter
+import Prometheus
```

Comment:
I would either make this a qualified import, or explicitly import the functions and types used from this module.

Reply:
True, I changed it to explicit imports now. We should do that everywhere in IP eventually.